packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -199,17 +211,50 @@ export function getDefaultProvider(input: FieldProvider): DefaultProvider {
 	return input as unknown as DefaultProvider;
 }
 
+/**
+ * Default field schema metadata shape.
+ *
+ * @remarks
+ *
+ * All properties are strictly optional.
+ * These particular properties exist both as an example of what kinds of data can be associated with a field,
+ * and may also be used by the system and related tooling in some cases.
+ *
+ * @sealed
+ * @public
+ */
+export interface FieldSchemaMetadata<
+	TCustomMetadata extends Record<string, unknown> = Record<string, unknown>,
+> {
+	/**
+	 * User-defined metadata.
+	 */
+	custom?: TCustomMetadata;
+
+	/**
+	 * The description of the field.
+	 *
+	 * @remarks
+	 *
+	 * If provided, will be used by the system in scenarios where a description of the field is useful.
+	 * E.g., when converting a node schema to {@link https://json-schema.org/ | JSON Schema}, this description will be
+	 * used as the `description` field.
+	 */
+	description?: string | undefined;
+}
+
 /**
  * Package internal construction API.
  */
 export let createFieldSchema: <
 	Kind extends FieldKind = FieldKind,
 	Types extends ImplicitAllowedTypes = ImplicitAllowedTypes,
+	TCustomMetadata extends Record<string, unknown> = Record<string, unknown>,
```

Comment:
I'm curious for these generic defaults (and similar elsewhere) - is there a case for them defaulting to `never` rather than to an empty record? If there are no customs, then the bag need not exist right? Though maybe it's easier to just have an empty bag.

Reply:
That pattern doesn't work out well. The inference in the schema factory methods doesn't work nicely, and the metadata is the third type parameter, so we would be forcing the user to specify the first two defaults when providing custom metadata :/